core/domain/learner_progress_services.py

```diff
@@ -443,6 +566,47 @@ def _remove_activity_ids_from_playlist(
         learner_playlist_services.save_learner_playlist(learner_playlist)
 
 
+def remove_story_from_completed_list(user_id, story_id):
```

Comment:
When would this function be called in the userflow? Since, i thought once a story was completed, it was never changed. If this is just called as part of another function, see where that function is linked to, in the frontend and check whether that would be called for stories or topics.

Reply:
This function is used to remove a completed story when some new chapter has been added to it. In that case, the story will be removed from the completed section. So before rendering the data to the user, we filter out the stories which are no longer present and the ones in which new chapters have been added from the completed list using this function.

Reply:
Ah ok!